### ISSUE TITLE
SSL connection errors on Windows when using cl+ssl:make-context

### ISSUE THREAD

**Issue by mtstickney** [cl-plus-ssl/cl-plus-ssl#50]
Connecting to a server with `cl+ssl:make-ssl-client-stream` by itself works correctly, but creating a context and wrapping the call in `cl+ssl:with-global-context` causes the connection to error out (at least on CCL -- on SBCL, it hangs waiting for the connection to become readable). This happens on Windows 7, although there's one [report](https://github.com/fukamachi/dexador/issues/33#issuecomment-300043472) that it _doesn't_ happen on Windows 10 (I'll try to confirm this after work).

This causes problems for drakma tip, and the current quicklisp version of dexador (probably others too).

Test case:
````
(ql:quickload '(:cl+ssl :usocket))

(defun connect (host port)
  (let ((sock (usocket:socket-connect host port))
        (ctx (cl+ssl:make-context :verify-mode cl+ssl:+ssl-verify-none+
                                  :verify-location :default)))
    (cl+ssl:with-global-context (ctx :auto-free-p t)
      (cl+ssl:make-ssl-client-stream (usocket:socket-stream sock)
                                    :hostname host))))

(defun connect-stream (host port)
  (let ((sock (usocket:socket-connect host port)))
    (cl+ssl:make-ssl-client-stream (usocket:socket-stream sock)
                                  :hostname host)))

(connect "github.com" 443)
;; > Error: The TLS/SSL connection on handle #<A Foreign Pointer #x2363D60> has been closed (return code: 5).
;; >        SSL error queue is empty.

(connect-stream "github.com" 443)
;; #<SSL-STREAM for #<BASIC-TCP-STREAM ISO-8859-1 (SOCKET/1976) #xD01FFCE>>
````

cl+ssl tip (50a857368a5fd9d54c02ff2bcb50f406a1b81250)
CCL version 1.11-r16635  (WindowsX8632)
OpenSSL version 1.0.2l
SBCL versions 1.3.9 and 1.3.12

**Comment by mtstickney** [comment]
I can confirm that this does _not_ occur on Windows 10 under SBCL 1.3.9 and cl+ssl tip. At this point I'm not sure if this is a problem with cl+ssl or OpenSSL; it's gonna need more investigation.

**Comment by avodonosov** [comment]
Thanks for the report. I, unfortunately, don't have Windows and also need to fix #46 first.

If `cl+ssl:make-ssl-client-stream`, and `cl+ssl:make-context` fails the problem should be easy to track down for someone able to reproduce by looking at the differences between `cl+ssl::*default-context*` initialization and what `cl+ssl:make-context` does; and trying changes to determine exactly what difference is responsible to the problem.

Pull requests and explanation would be much appreciated.

**Comment by mtstickney** [comment]
After a bit of testing, I've at least identified the culprit: when the `:method` arg isn't passed to `cl+ssl:make-context`, the disabled protocls list is set to `(list +SSL-OP-NO-SSLV2+ +SSL-OP-NO-SSLV3+)`, which causes the error (I've confirmed this directly by passing that list as the `:disabled-protocols` argument).

Still not sure why that causes a connection problem on Windows 7 but not Windows 10; I'll keep investigating.
